### PR TITLE
Alex sichitiu 01 db init

### PR DESCRIPTION
**Pull request by AlexSichitiu**
Initialized database and Prisma Client.

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+generator client {
+  provider = "prisma-client-js"
+}
+
+model User {
+  id              String    @id
+  email           String    @unique
+  hashed_password String?
```
Comment (LukarioMC) [inline prisma/schema.prisma]: Isn't password required if they have an account?

File: prisma/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  fname           String?
+  lname           String?
+  is_admin        Boolean   @default(false)
+  Session         Session[]
```
Comment (LukarioMC) [inline prisma/schema.prisma]: There should be only one active or expired session per user, would this potentially allow for multiple?

File: prisma/schema.prisma [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  Factoid         Factoid[]
+  Report          Report[]
+}
+
+model Session {
+  id          Int      @id @default(autoincrement())
+  user        User?    @relation(fields: [user_id], references: [id])
+  user_id     String?
```
Comment (LukarioMC) [inline prisma/schema.prisma]: I believe sessions wouldn't be tracking non-logged in accounts, so user/id should be required in this case?

File: prisma/schema.prisma [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  category_id Int
+
+  @@id([factoid_id, category_id])
+}
+
+model Attachment {
+  id         Int     @id @default(autoincrement())
+  factoid    Factoid @relation(fields: [factoid_id], references: [id])
+  factoid_id Int
+  link       String?
```
Comment (LukarioMC) [inline prisma/schema.prisma]: Should link be required for an attachment?

File: prisma/schema.prisma (hunk 3 above)
Comment (AlexSichitiu) [reply prisma/schema.prisma]: You're right, this was a copy-paste error.

File: prisma/schema.prisma (hunk 1 above)
Comment (AlexSichitiu) [reply prisma/schema.prisma]: You're right, this was a carryover from when we were considering doing public identifiers based on email only.

File: prisma/schema.prisma (hunk 2 above)
Comment (AlexSichitiu) [reply prisma/schema.prisma]: You're right, I'm looking into implementing 1:1 relationships in Prisma.

File: prisma/schema.prisma (hunk 4 above)
Comment (AlexSichitiu) [reply prisma/schema.prisma]: Making this field non-nullable.

**Comment by LukarioMC** [comment]
I have some questions still but they are more for Tom to answer. So as it is, this looks good as a DB base for now.

If we require any changes Prisma should be able to simply migrate. I approve.

**Review by LukarioMC — APPROVED** [review]

**Pull request merged**